pkg/chunk/chunk_store.go

```diff
@@ -343,6 +353,42 @@ func (c *store) getMetricNameChunks(ctx context.Context, userID string, from, th
 
 	// Filter out chunks based on the empty matchers in the query.
 	filteredChunks := filterChunksByMatchers(allChunks, filters)
+	return c.applyTombstonesToChunks(userID, filteredChunks, from, through)
+}
+
+// filters out completely deleted chunks and sets deletedIntervals for partially deleted chunks
+func (c *store) applyTombstonesToChunks(userID string, chunks []Chunk, queryRangeFrom, queryRangeThrough model.Time) ([]Chunk, error) {
```

Comment:
What impact will this have on query performance? If there is a large number of deletion jobs will that have a significant impact? If so we may want to consider isolating tombstones are per user basis.

Reply:
Yes, there would be impact due to this on the query side. After discussing this in community call we are okay with it given it is only going to impact users who have pending delete requests.